### ISSUE TITLE
addDocumentToNamespace undefined

### ISSUE THREAD

**Issue by phires** [Mintplex-Labs/anything-llm#197]
Hi,

after uploading a document I want to active it for the workspace.
The Logfile shows the following error:

`2023-08-17 09:51:09 SELECT * FROM system_settings WHERE label = 'multi_user_mode'
2023-08-17 09:51:09 SELECT * FROM workspaces WHERE slug = 'privat-test'
2023-08-17 09:51:09 SELECT * FROM workspace_documents WHERE workspaceId = 1 
2023-08-17 09:51:09 Adding new vectorized document into namespace privat-test
2023-08-17 09:51:09 Chunks created from document: 4
2023-08-17 09:51:09 addDocumentToNamespace undefined
2023-08-17 09:51:09 Failed to vectorize test-doc-1692258645/test-doc-pg0-228645c0-8078-4dbd-bc63-a96cbc9c8dcd.json
2023-08-17 09:51:09 Adding new vectorized document into namespace privat-test
2023-08-17 09:51:09 Chunks created from document: 3
2023-08-17 09:51:09 addDocumentToNamespace undefined
2023-08-17 09:51:09 Failed to vectorize test-doc-1692258645/test-doc-pg1-2b3b980f-9c53-4bd9-b7bc-1b28f0807e5a.json
2023-08-17 09:51:09 Adding new vectorized document into namespace privat-test
2023-08-17 09:51:09 Chunks created from document: 1
2023-08-17 09:51:09 addDocumentToNamespace undefined`

I am using docker.
ChromaDB is also a docker container.
Disabling Vector Cache does not help.

Anything I can do?

**Comment by igadalla** [comment]
![image](https://github.com/Mintplex-Labs/anything-llm/assets/122039966/46dfd9d5-4ceb-4ac9-844d-ce22e37b804c)


I also have the same error, but with Pinecone and using Docker.

- addDocumentToNamespace PineconeClient: Error calling upsert: PineconeError: The requested feature 'Namespaces' is not supported by the current index type 'Starter'
- Failed to vectorize file-to-test-1692280064/file-to-test-pg0-2af44f9e-ee22-4774-a934-f1e6ad6d4394.json

any help :)

**Comment by timothycarambat** [comment]
> Hi,
> 
> after uploading a document I want to active it for the workspace. The Logfile shows the following error:
> 
> ```
> 2023-08-17 09:51:09 SELECT * FROM system_settings WHERE label = 'multi_user_mode'
> 2023-08-17 09:51:09 SELECT * FROM workspaces WHERE slug = 'privat-test'
> 2023-08-17 09:51:09 SELECT * FROM workspace_documents WHERE workspaceId = 1 
> 2023-08-17 09:51:09 Adding new vectorized document into namespace privat-test
> 2023-08-17 09:51:09 Chunks created from document: 4
> 2023-08-17 09:51:09 addDocumentToNamespace undefined
> 2023-08-17 09:51:09 Failed to vectorize test-doc-1692258645/test-doc-pg0-228645c0-8078-4dbd-bc63-a96cbc9c8dcd.json
> 2023-08-17 09:51:09 Adding new vectorized document into namespace privat-test
> 2023-08-17 09:51:09 Chunks created from document: 3
> 2023-08-17 09:51:09 addDocumentToNamespace undefined
> 2023-08-17 09:51:09 Failed to vectorize test-doc-1692258645/test-doc-pg1-2b3b980f-9c53-4bd9-b7bc-1b28f0807e5a.json
> 2023-08-17 09:51:09 Adding new vectorized document into namespace privat-test
> 2023-08-17 09:51:09 Chunks created from document: 1
> 2023-08-17 09:51:09 addDocumentToNamespace undefined
> ```
> 
> I am using docker. ChromaDB is also a docker container. Disabling Vector Cache does not help.
> 
> Anything I can do?

What URL are you using to connect to Chroma? if both are dockerized you may need to use the docker-gateway in anythingLLM of `http://host.docker.internal:[CHROMA_PORT]` since they are both in docker and the AnythingLLM docker container will look internal for localhost:8000 as opposed to the host machine

**Comment by timothycarambat** [comment]
@igadalla this error results from a pinecone instance that is not in the ready state. Is your index name spelled properly? Last time I saw this error the user had correct API keys, but the index name was incorrect.

**Comment by phires** [comment]
> > Hi,
> > after uploading a document I want to active it for the workspace. The Logfile shows the following error:
> > ```
> > 2023-08-17 09:51:09 SELECT * FROM system_settings WHERE label = 'multi_user_mode'
> > 2023-08-17 09:51:09 SELECT * FROM workspaces WHERE slug = 'privat-test'
> > 2023-08-17 09:51:09 SELECT * FROM workspace_documents WHERE workspaceId = 1 
> > 2023-08-17 09:51:09 Adding new vectorized document into namespace privat-test
> > 2023-08-17 09:51:09 Chunks created from document: 4
> > 2023-08-17 09:51:09 addDocumentToNamespace undefined
> > 2023-08-17 09:51:09 Failed to vectorize test-doc-1692258645/test-doc-pg0-228645c0-8078-4dbd-bc63-a96cbc9c8dcd.json
> > 2023-08-17 09:51:09 Adding new vectorized document into namespace privat-test
> > 2023-08-17 09:51:09 Chunks created from document: 3
> > 2023-08-17 09:51:09 addDocumentToNamespace undefined
> > 2023-08-17 09:51:09 Failed to vectorize test-doc-1692258645/test-doc-pg1-2b3b980f-9c53-4bd9-b7bc-1b28f0807e5a.json
> > 2023-08-17 09:51:09 Adding new vectorized document into namespace privat-test
> > 2023-08-17 09:51:09 Chunks created from document: 1
> > 2023-08-17 09:51:09 addDocumentToNamespace undefined
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I am using docker. ChromaDB is also a docker container. Disabling Vector Cache does not help.
> > Anything I can do?
> 
> What URL are you using to connect to Chroma? if both are dockerized you may need to use the docker-gateway in anythingLLM of `http://host.docker.internal:[CHROMA_PORT]` since they are both in docker and the AnythingLLM docker container will look internal for localhost:8000 as opposed to the host machine

Yes, I am using http://host.docker.internal:8000 as Chroma URL.

**Comment by timothycarambat** [comment]
@phires Earlier this week we improved errors on this function - can you pull in latest and replicate the error. The logs should now show a verbose error for when this happens which will make tracing it much easier!

**Comment by igadalla** [comment]
> @igadalla this error results from a pinecone instance that is not in the ready state. Is your index name spelled properly? Last time I saw this error the user had correct API keys, but the index name was incorrect.

Yes, the Index name is correct. I checked that by changing it once, and didn't work. so yeah, the index name is 100% correct, and the problem is, when I try to upload a document. The whole server refreshes and starts from the beginning. And I get the error in Docker: "Failed to vectorize file."

**Comment by phires** [comment]
> @phires Earlier this week we improved errors on this function - can you pull in latest and replicate the error. The logs should now show a verbose error for when this happens which will make tracing it much easier!

I was able to solve the issue with the improved error messages.
It was related to my OpenAI Deployment in Azure: `Error: Azure OpenAI Failed to embed: (1) Embedding Errors! [undefined]: The embeddings operation does not work with the specified model, gpt-35-turbo. Please choose different model and try again. You can learn more about which models can be used with each operation here: https://go.microsoft.com/fwlink/?linkid=2197993.`
After redeploying the embedding model it is now working!
Thanks for your help :)
